### ISSUE TITLE
migration_diff warning `Undefined array key "precision"`

### ISSUE THREAD

**Issue by Rakasch** [cakephp/migrations#649]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.4.18
* Migrations plugin version: 3.9.0
* Bake plugin version (if relevant): 2.9.3
* Database server (MySQL): 10.6.12-MariaDB-0ubuntu0.22.04.1
* PHP Version: 8.1.22
* Platform / OS: Ubuntu 22.04.3 LTS

### What you did
Tried to create a migration file with
> bin/cake bake migration_diff ModStCommentsUpdateDeletedBool

I made only one change to the database table.
I changed the datatype from tinyint(4) to tinyint(1).
Before:
![image](https://github.com/cakephp/migrations/assets/8471027/127d0801-cbd1-41e8-9715-6a687fa559ff)
After:
![image](https://github.com/cakephp/migrations/assets/8471027/4a08658e-4e25-49cd-a200-44b5a2fe9b58)

### Expected Behavior
create migration file without warning

### Actual Behavior
displays a warning about `Undefined array key "precision"`
Despite the warning, the migration was apparently created correctly.
![image](https://github.com/cakephp/migrations/assets/8471027/49fb79d5-db70-49e1-ad22-30e461888917)

![image](https://github.com/cakephp/migrations/assets/8471027/8ba97965-64f5-41b7-ac96-d7606d45d346)

**Comment by dereuromark** [comment]
Is this still valid? Looks a bit weird when there is only int involved.

**Comment by Rakasch** [comment]
yes it still happens (I'm on cakephp 4.5.1, with migrations 3.9.0)
![image](https://github.com/cakephp/migrations/assets/8471027/809e0e71-b781-4cca-8028-eed43527ce9c)

**Comment by LordSimal** [comment]
I am unable to reproduce this (neither with the `builtin` backend nor the old phinx backend) in current CakePHP 5.2.5 and Migrations 4.6.4
